src/arglist.cpp

```diff
@@ -14,6 +14,12 @@ ArgList::ArgList(const std::string &s, char delim) {
     reset();
 }
 
+ArgList::ArgList(Container::const_iterator from, Container::const_iterator to)
+{
+    c_ = std::make_shared<Container>(from, to);
```

Comment:
Not really related to this PR, but I think that this `c_` member deserves a meaningful name. Even calling it `data_` (still very generic) would be an improvement.

Reply:
`container_`?

Reply:
Yup, sounds good to me!